### ISSUE TITLE
pre-commit: golangci-lint uses wrong JSON schema (v2.0), causing config verification errors

### ISSUE THREAD

**Issue by gnought** [golangci/golangci-lint#6347]
### Welcome

- [x] Yes, I'm using a binary release within 2 latest [releases](https://github.com/golangci/golangci-lint/releases). Only such installations are supported.
- [x] Yes, I've searched similar [issues on GitHub](https://github.com/golangci/golangci-lint/issues) and didn't find any.
- [x] Yes, I've read the `typecheck` section of the [FAQ](https://golangci-lint.run/docs/welcome/faq/#why-do-you-have-typecheck-errors).
- [x] Yes, I've tried with the standalone [linter](https://golangci-lint.run/docs/linters/) if available (e.g., gocritic, go vet, etc.).
- [x] I agree to follow this project's [Code of Conduct](https://github.com/golangci/golangci-lint?tab=coc-ov-file)

### How did you install golangci-lint?

Other

### Description of the problem

I am using **pre-commit** with golangci-lint pinned to v2.8.0.
However, the golangci-lint-config-verify hook fails with JSON schema validation errors.

Upon investigation, I noticed that the golangci-lint binary installed inside the pre-commit cache reports its version as:

`2.0.0-20260107212322-e2e40021c900+dirty`

Because of this, golangci-lint fetches the v2.0 JSON schema:
> https://golangci-lint.run/jsonschema/golangci.v2.0.jsonschema.json

This schema is incompatible with newer configuration options supported in v2.8.0, which leads to false validation failures.

**Steps to Reproduce**

.pre-commit-config.yaml:

```yaml
---
default_stages:
  - pre-commit
  - pre-push

repos:
  - repo: https://github.com/golangci/golangci-lint
    rev: v2.8.0
    hooks:
      - id: golangci-lint-config-verify
      - id: golangci-lint-fmt
      - id: golangci-lint-full

```

Run:
`pre-commit run -a`

**Actual Result**
```bash
golangci-lint-config-verify..............................................Failed
- hook id: golangci-lint-config-verify
- exit code: 3

jsonschema: "linters.settings.exhaustruct" does not validate with "/properties/linters/properties/settings/properties/exhaustruct/$ref/additionalProperties":
additional properties 'allow-empty-returns' not allowed

jsonschema: "linters.settings" does not validate with "/properties/linters/properties/settings/additionalProperties":
additional properties 'embeddedstructfieldcheck', 'funcorder', 'godoclint' not allowed
```

The command is terminated due to an error: the configuration contains invalid elements

Meanwhile, other hooks succeed:

```
golangci-lint-fmt  Passed
golangci-lint-full Passed
```

Observed Binary Version in Pre-commit Cache
```
~/.cache/pre-commit/repom7p7wvum/golangenv-system/bin/golangci-lint --version

golangci-lint has version 2.0.0-20260107212322-e2e40021c900+dirty
built with go1.25.7
from e2e40021c9007020676c93680a36e3ab06c6cd33 (modified: true)
built at 2026-01-07T21:23:22Z
```

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version 2.0.0-20260107212322-e2e40021c900+dirty
built with go1.25.7
from e2e40021c9007020676c93680a36e3ab06c6cd33 (modified: true)
built at 2026-01-07T21:23:22Z
```

</details>

### Configuration

<details>

```console
# paste configuration file or CLI flags here
```

</details>

### Go environment

<details>

```console
$ go version && go env
# paste output here
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
# paste output here
```

</details>

### A minimal reproducible example or link to a public repository

<details>

```go
// add your code here
```

</details>

### Validation

- [x] Yes, I've included all information above (version, config, etc.).

### Supporter

- [ ] I am a sponsor/backer through [GitHub](https://github.com/sponsors/golangci/) or [OpenCollective](https://opencollective.com/golangci-lint)

**Comment by ldez** [comment]
You are using:
```
golangci-lint has version 2.0.0-20260107212322-e2e40021c900+dirty
```

So the JSON schema is the `golangci.v2.0.jsonschema.json` and not the one related to v2.8.0.

Please update your golangci-lint version to v2.8.0

**Comment by ldez** [comment]
I don't know how pre-commit is defining the version of the binary but the version is wrong.
The commit is right but not the version.

**Comment by ldez** [comment]
This is because pre-commit is using a dangling commit without tag information.

VCS metadata are not available, and it seems like pre-commit doesn't want to change that or add an option about it.

**Comment by ilovelinux** [comment]
Same behavior happens with [`prek`](https://github.com/j178/prek).

**Comment by ccoVeille** [comment]
> Same behavior happens with [`prek`](https://github.com/j178/prek).

Prek is active,  they may accept to work on fixing this
